cli/admin/src/main/java/io/pravega/cli/admin/controller/ControllerCommand.java

```diff
@@ -47,14 +58,39 @@
      * @return REST client.
      */
     protected Context createContext() {
+        CLIControllerConfig config = getCLIControllerConfig();
         ClientConfig clientConfig = new ClientConfig();
         clientConfig.register(JacksonJsonProvider.class);
         clientConfig.property("sun.net.http.allowRestrictedHeaders", "true");
-        Client client = ClientBuilder.newClient(clientConfig);
+
+        Client client;
+
+        // If TLS parameters are configured, set them in client.
+        if (config.isTlsEnabled()) {
+            SSLContext tlsContext;
+            try {
+                KeyStore ks = createTrustStore(config.getTruststore());
+                TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                tmf.init(ks);
+                tlsContext = SSLContext.getInstance("TLS");
+                tlsContext.init(null, tmf.getTrustManagers(), null);
+            } catch (KeyStoreException | IOException | NoSuchAlgorithmException | CertificateException | KeyManagementException e) {
+                output("Encountered exception while trying to use the given truststore: %s", config.getTruststore());
```

Comment:
Without the additional context, it'd be very difficult to debug this issue. Please output the exception message/trace too. 

Curious, why don't we add some logs to these methods too?

Reply:
I print the error message.